text/0059-token-metadata-document.md

```diff
@@ -0,0 +1,504 @@
+* `ticker_symbol`: A ticker symbol for the currency. This is appropriate to be displayed by an exchange, as a short form of the name of the currency. Not more than 12 printable ASCII characters, including `[A-Za-z0-9].-`.
+* `symbol`: Most fiat currencies have a printable character such as $, £, ¥. Some cryptocurrencies do also, Bitcoin has ₿, and Ethereum has Ξ. This may be optionally specified as a UTF-8 string in the "symbol" field. How exactly it is displayed may be locale specific and we do not attempt to formally specify that at this time.
+* `decimals`: An integer specifying how the `u64` integer in a TxOut in the blockchain is scaled to compute a user-displayed amount of the currency.
+* `logo_svg`: An optional logo image. This is a base64-encoded SVG document. This is expected to have been sanitized using something like [svg-hush](https://github.com/cloudflare/svg-hush), to remove scripting, hyperlinks to other documents, and references to cross-origin resources. The full extent of such sanitization will not be specified here.
+* `info_url`: A link to a website containing more information about the token that may be interesting to token holders. This should basic information about the purpose of the token, its supply, any utility that it has, or links to associated whitepapers.
+
+Clients MUST download and validate the `token_metadata.sig` before attempting to process the `token_metadata.json`, and must reject the json with an error if the signature fails.
+
+Clients SHOULD include a minimum for `signing_timestamp` as part of their build, and reject any `token_metadata.json` which is less than the baked-in `signing_timestamp`. This prevents replay attacks where an old `token_metadata.json` document is substituted for the latest one with the goal of confusing the users by making their tokens display differently or not at all. They can fetch the latest `token_metadata.json` at build time to get the latest `signing_timestamp`.
+
+Clients MAY continue to bake token metadata into their build at their discretion, for instance, in order to support offline wallets or other cases where the client cannot access the internet. When both baked-in metadata and the `token_metadata.json` are available, the validated `token_metadata.json` should be considered a more reliable source of truth.
+
+Maintainers of this document SHOULD NOT:
+
+* Allow duplicate records for a given token id
+* Delete a token id's record
+* Modify data such as the `ticker_symbol` or `decimals` of a token, which may confuse users, and particularly, any exchanges that use this as a source of truth.
+* There may be legitimate reasons to make changes the `currency_name` or `logo`, but this needs to be carefully considered.
+* Allow two token ids with `currency_name`s which could be confused
+* Allow the same `ticker_symbol` to appear, or, allowing two `ticker_symbols` which differ only in case, or the replacement of letters with similar numbers.
```

Comment:
:question: Can this restriction really be enforced at all with in the `token_metadata.json`?
If an exchange decides to post `M0B` and the foundation decides that it needs to be in the block chain, is there some way to place an alternate symbol here?

```suggestion
* Allow the same `ticker_symbol` to appear, or, allowing two `ticker_symbol`s which differ only in case, or the replacement of letters with similar numbers.
```

Reply:
We could rename from "ticker symbol" to "short code", that matches shopify docu around fiat currencies.

My expectation is that whatever we put here will be suggestive of what an exchange uses as ticker symbol (and vice versa).

The thing is, shopify calls it an ISO short code, and I am not sure that cryptocurrencies have ISO short codes the same way fiat currencies do. But maybe this field can just be a non- specific short-code.